tests/e2e/Services/Sites/SitesCustomServerTest.php

```diff
@@ -939,5 +940,68 @@ public function testGetFrameworks(): void
         $this->assertArrayHasKey('adapters', $framework);
     }
 
+    public function testConsoleAvailabilityEndpoint(): void
+    {
+        $site = $this->createSite([
+            'buildRuntime' => 'ssr-22',
+            'fallbackFile' => null,
+            'framework' => 'other',
+            'name' => 'Test Site',
+            'subdomain' => 'test-site',
+            'outputDirectory' => './',
+            'siteId' => ID::unique()
+        ]);
+
+        $siteId = $site['body']['$id'] ?? '';
+
+        $this->assertEquals(201, $site['headers']['status-code']);
+        $this->assertNotEmpty($site['body']['$id']);
+        $this->assertEquals('Test Site', $site['body']['name']);
+
+        $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
+        $domain = "test-site.{$sitesDomain}";
+        $ruleId1 = \md5($domain);
+
+        $response = $this->client->call(Client::METHOD_GET, '/console/resources/' . $ruleId1, [
+            'origin' => 'http://localhost',
+            'content-type' => 'application/json',
+            'cookie' => 'a_session_console=' . $this->getRoot()['session'],
+            'x-appwrite-project' => 'console',
+        ], [
+            'type' => 'rules',
+        ]);
+
+        $this->assertEquals(409, $response['headers']['status-code']); // subdomain unavailable
+
+        $domain = "non-existent-subdomain.{$sitesDomain}";
+        $ruleId2 = \md5($domain);
+
+        $response = $this->client->call(Client::METHOD_GET, '/console/resources/' . $ruleId2, [
+            'origin' => 'http://localhost',
+            'content-type' => 'application/json',
+            'cookie' => 'a_session_console=' . $this->getRoot()['session'],
+            'x-appwrite-project' => 'console',
+        ], [
+            'type' => 'rules',
+        ]);
```

Comment:
If we generate rule IDs in tests, it means we will need to generate them in Console, and doing md5 hash in Console doesn't sound right.
Can we accept domain directly for the availability check endpoint instead?